src/nft/components/sell/list/ListPage.tsx

```diff
@@ -1,3 +1,841 @@
+            ROYALTIES
+          </Column>
+          <Column
+            className={badge}
+            color="darkGray"
+            style={{ flex: '1.5' }}
+            display={{ sm: 'none', md: 'flex' }}
+            textAlign="right"
+          >
+            YOU RECEIVE
+          </Column>
+        </Row>
+      </Row>
+      {sellAssets.map((asset) => {
+        return (
+          <>
+            <NFTListRow
+              asset={asset}
+              globalPriceMethod={globalPriceMethod}
+              globalPrice={globalPrice}
+              setGlobalPrice={setGlobalPrice}
+              selectedMarkets={selectedMarkets}
+            />
+            {sellAssets.indexOf(asset) < sellAssets.length - 1 && <hr className={styles.nftDivider} />}
+          </>
+        )
+      })}
+    </Column>
+  )
+}
+
+enum WarningType {
+  BELOW_FLOOR = 'LISTING BELOW FLOOR ',
+  ALREADY_LISTED = 'ALREADY LISTED FOR ',
+  NONE = '',
+}
+
+const PriceTextInput = ({
+  listPrice,
+  setListPrice,
+  isGlobalPrice,
+  setGlobalOverride,
+  globalOverride,
+  warning,
+  asset,
+}: any) => {
+  const [focused, setFocused] = useState(false)
+  const [warningType, setWarningType] = useState(WarningType.NONE)
+  const removeMarketplaceWarning = useSellAsset((state) => state.removeMarketplaceWarning)
+  const removeSellAsset = useSellAsset((state) => state.removeSellAsset)
+
+  useEffect(() => {
+    setWarningType(WarningType.NONE)
+    if (!warning && listPrice) {
+      if (listPrice < asset.floorPrice) setWarningType(WarningType.BELOW_FLOOR)
+      else if (asset.floor_sell_order_price && listPrice >= asset.floor_sell_order_price)
+        setWarningType(WarningType.ALREADY_LISTED)
+    } else if (warning && !isNaN(listPrice) && listPrice >= 0) removeMarketplaceWarning(asset, warning)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [listPrice])
+
+  return (
+    <Column gap="12" position="relative">
+      <Row
+        color="medGray"
+        height="44"
+        width="min"
+        padding="4"
+        borderRadius="8"
+        style={{
+          border: `1px solid ${
+            warning && !focused
+              ? vars.color.orange
+              : isGlobalPrice
+              ? vars.color.genieBlue
+              : listPrice != null && listPrice !== ''
+              ? themeVars.colors.darkGray
+              : vars.color.grey700
+          }`,
+        }}
+      >
+        <NumericInput
+          as="input"
+          pattern="[0-9]"
+          borderStyle="none"
+          className={body}
+          color={{ placeholder: 'darkGray', default: 'blackBlue' }}
+          placeholder="Set"
+          marginRight="0"
+          marginLeft="14"
+          backgroundColor="none"
+          style={{ width: '68px' }}
+          onFocus={() => setFocused(true)}
+          onBlur={() => {
+            setFocused(false)
+          }}
+          value={listPrice || ''}
```

Comment:
if listPrice is 0 would we want it to show up as 0 or '', in this case its going to show as '' but if its `?? ''` instead of `|| ''` it will show up as 0

Reply:
did a major refactor of types, see if it's more clear now